src/array.rs

```diff
@@ -117,9 +117,16 @@ pub type PyArray6<T> = PyArray<T, Ix6>;
 /// Dynamic-dimensional array.
 pub type PyArrayDyn<T> = PyArray<T, IxDyn>;
 
+static ARRAY_MODULE_CELL: GILOnceCell<Py<PyModule>> = GILOnceCell::new();
+
 /// Returns a handle to NumPy's multiarray module.
 pub fn get_array_module(py: Python<'_>) -> PyResult<&PyModule> {
-    PyModule::import(py, npyffi::array::MOD_NAME)
+    let module = ARRAY_MODULE_CELL
```

Comment:
I think we misunderstood here: The idea is to cache the `asarray` attribute of the NumPy module, not the module reference itself. This is both more efficient and it avoids changing the performance characteristics of `get_array_module`.

Reply:
You are right, I misunderstood. I'll revert this in my next commit.